app/src/main/java/nerd/tuxmobil/fahrplan/congress/details/SessionDetailsFragment.kt

```diff
@@ -243,19 +291,27 @@ class SessionDetailsFragment : Fragment(), SessionDetailsViewModel.ViewActionHan
         this.isVisible = true
     }
 
+    private fun updateOptionsMenu() {
+        requireActivity().invalidateOptionsMenu()
+    }
+
     override fun onCreateOptionsMenu(menu: Menu, inflater: MenuInflater) {
         super.onCreateOptionsMenu(menu, inflater)
+        if (!::model.isInitialized) {
+            // Skip if lifecycle is faster than ViewModel.
+            return
+        }
         inflater.inflate(R.menu.detailmenu, menu)
-        if (viewModel.isFlaggedAsFavorite) {
+        if (model.isFlaggedAsFavorite) {
             menu.findItem(R.id.menu_item_flag_as_favorite)?.let { it.isVisible = false }
             menu.findItem(R.id.menu_item_unflag_as_favorite)?.let { it.isVisible = true }
```

Comment:
Just a thought:
you could make these a bit more nicer, in my opinion with another extension function:

```
private fun Menu.setMenuItemVisibility(itemId: Int, visibility: Boolean) {
    findItem(itemId)?.let { item -> item.isVisible = visibility }
}
 ```

Reply:
:heavy_check_mark: Adapted in 705739fa159e2ce83615bf013a493b145b39d48a.